chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -543,14 +546,47 @@
                   target_platform_version: target_platform_version)
       end
 
-      missing_files = !File.exist?("#{os_dir}/install/boot/#{arch}/common")
+      if target_platform_distro == "suse"
+        missing_files = !File.exist?("#{os_dir}/install/boot/#{arch}/common")
+      elsif target_platform_distro == "opensuse"
+        # if we use remote URL, then we don't require a rsync of the install
+        # media, but we still need to copy the kernel/initrd
+        install_url = params["install_url"]
+
+        missing_files = !File.exist?("#{os_dir}/install/boot/#{arch}/common")
+
+        if missing_files && !install_url.nil? && !install_url.empty?
+          kernel_path = Pathname.new(os_dir).join("install").join(kernel)
+          kernel_tmp_path = Pathname.new(os_dir).join("install").join("#{kernel}.check")
+          initrd_path = Pathname.new(os_dir).join("install").join(initrd)
+          initrd_tmp_path = Pathname.new(os_dir).join("install").join("#{initrd}.check")
+
+          unless kernel_path.exist?
+            kernel_path.dirname.mkpath
+            `wget --timeout 5 -O #{kernel_tmp_path} #{install_url}/#{kernel}`
+            if Digest::MD5.file(kernel_tmp_path.to_s).hexdigest == "fcc6dc33eac95bdf96c6cea4dd3eb7c2"
+              kernel_tmp_path.rename(kernel_path.to_s)
+            end
+          end
+
+          unless initrd_path.exist?
+            initrd_path.dirname.mkpath
+            `wget --timeout 5 -O #{initrd_tmp_path} #{install_url}/#{initrd}`
+            if Digest::MD5.file(initrd_tmp_path.to_s).hexdigest == "c15afdcd69e1866c6df724d559aabe51"
+              initrd_tmp_path.rename(initrd_path.to_s)
+            end
+          end
+
+          missing_files = !kernel_path.exist? || !initrd_path.exist?
+        end
+      end
 
     when /^(redhat|centos)/ =~ os
       # Add base OS install repo for redhat/centos
       if ::File.exist? "#{tftproot}/#{os}/#{arch}/install/repodata"
-        node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=#{admin_web}" => true }
+        node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=#{install_url}" => true }
       else
-        node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=#{admin_web}/Server" => true }
+        node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=#{install_url}/Server" => true }
```

Comment:
Metrics/LineLength: Line is too long. [109/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)